src/main/java/com/iota/iri/network/spam/DropFixedPercentage.java

```diff
@@ -0,0 +1,80 @@
+package com.iota.iri.network.spam;
+
+import com.iota.iri.network.Neighbor;
+import com.iota.iri.utils.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Simple spam prevention strategy that considers a percentage of the most active neighbors to be spamming.
+ */
+public class DropFixedPercentage implements SpamPreventionStrategy {
+
+    // needs to be static according to IRI code style
+    private static final Logger LOG = LoggerFactory.getLogger(DropFixedPercentage.class);
+
+    private final int dropPercentage;
+    private Map<Neighbor, Long> transactionCount;
+    private Set<Neighbor> mostActiveNeighbors = Collections.emptySet();
+
+    /**
+     * Creates a new spam strategy that marks a fixed percentage of neighbors to be spamming.
+     * @param dropPercentage The percentage of neighbors that are considered to be spamming. Percentage will be rounded
+     *                       down. So 33% of 3 neighbors are 0 neighbors. 34% of 3 neighbors is one neighbor.
+     * @param neighbors The list of neighbors that should be examined for finding spamming nodes.
+     */
+    public DropFixedPercentage(int dropPercentage, List<Neighbor> neighbors) {
+        LOG.info("Created fixed percentage spam prevention strategy. Dropping {}% of most active neighbors.", dropPercentage);
+        this.dropPercentage = dropPercentage;
+        transactionCount = toMap(neighbors);
+    }
+
+    /**
+     * Recalculates what neighbors are currently spamming. The most active neighbors since the last call of this
+     * method will be marked to be spamming.
+     * @param neighbors The list of neighbors that should be examined for spamming neighbors.
+     */
+    public void calculateSpam(List<Neighbor> neighbors) {
+        List<Pair<Long, Neighbor>> deltas = new ArrayList<>(neighbors.size());
+        neighbors.forEach(neighbor -> {
+            Long previous = transactionCount.getOrDefault(neighbor, neighbor.getNumberOfAllTransactions());
+            long delta = neighbor.getNumberOfAllTransactions() - previous;
```

Comment:
according to CFB's explanation, this should count only the new transactions, not all:
```
[11:06] Come-from-Beyond: Take all *new* txs received from the neighbors during last 5 sec and count them. 
```

Reply:
Yes, fix incoming.